### PR TITLE
Apply black

### PR DESCRIPTION
**Pull request by suzil**
- Add pre-commit for Black
- Unpin dev requirements (pins not really needed here)
- Add Black badge

### CODE REVIEW

File: src/lieu/tfidf.py [hunk 1]
```diff
@@ -31,23 +31,20 @@ def update(self, doc):
...
-            'idf_counts': dict(self.idf_counts)
-        })
+        return json.dumps({"N": self.N, "idf_counts": dict(self.idf_counts)})
 
     def write(self, f):
-        writer = csv.writer(f, delimiter='\t')
+        writer = csv.writer(f, delimiter="\t")
         for k, v in self.idf_counts.items():
-            writer.writerow(['{}{}'.format(self.term_key_prefix, k), v])
+            writer.writerow(["{}{}".format(self.term_key_prefix, k), v])
```
Comment (ddelange) [inline src/lieu/tfidf.py]: f-string

File: src/lieu/coordinates.py [hunk 2]
```diff
@@ -141,17 +156,17 @@ def latlon_to_decimal(latitude, longitude):
...
     latitude = float(latitude)
     longitude = float(longitude)
 
     if not is_valid_latitude(latitude):
-        raise ValueError('Invalid latitude: {}'.format(latitude))
+        raise ValueError("Invalid latitude: {}".format(latitude))
 
     if not is_valid_longitude(longitude):
-        raise ValueError('Invalid longitude: {}'.format(longitude))
+        raise ValueError("Invalid longitude: {}".format(longitude))
```
Comment (ddelange) [inline src/lieu/coordinates.py]: f

File: src/lieu/coordinates.py [hunk 3]
```diff
@@ -141,17 +156,17 @@ def latlon_to_decimal(latitude, longitude):
...
-        longitude = re.sub(end_re, '', longitude)
+        longitude = re.sub(beginning_re, "", longitude)
+        longitude = re.sub(end_re, "", longitude)
 
     latitude = float(latitude)
     longitude = float(longitude)
 
     if not is_valid_latitude(latitude):
-        raise ValueError('Invalid latitude: {}'.format(latitude))
+        raise ValueError("Invalid latitude: {}".format(latitude))
```
Comment (ddelange) [inline src/lieu/coordinates.py]: f

File: src/lieu/coordinates.py [hunk 4]
```diff
@@ -91,9 +106,9 @@ def to_valid_latitude(latitude):
 
 
 def to_valid_longitude(longitude):
-    '''Convert longitude into the -180 to 180 scale'''
+    """Convert longitude into the -180 to 180 scale"""
     if not is_valid_longitude(longitude):
-        raise ValueError('Invalid longitude {}'.format(longitude))
+        raise ValueError("Invalid longitude {}".format(longitude))
```
Comment (ddelange) [inline src/lieu/coordinates.py]: f

File: src/lieu/coordinates.py [hunk 5]
```diff
@@ -78,9 +93,9 @@ def is_valid_longitude(longitude):
 
 
 def to_valid_latitude(latitude):
-    '''Convert longitude into the -180 to 180 scale'''
+    """Convert longitude into the -180 to 180 scale"""
     if not is_valid_latitude(latitude):
-        raise ValueError('Invalid latitude {}'.format(latitude))
+        raise ValueError("Invalid latitude {}".format(latitude))
```
Comment (ddelange) [inline src/lieu/coordinates.py]: f

File: setup.py [hunk 6]
```diff
@@ -3,43 +3,43 @@
...
-            'Operating System :: MacOS :: MacOS X',
-            'Operating System :: POSIX :: Linux',
-            'Topic :: Text Processing :: Linguistic',
-            'Topic :: Scientific/Engineering :: GIS',
-            'Topic :: Software Development :: Libraries :: Python Modules'
+            "Intended Audience :: Developers",
+            "Intended Audience :: Information Technology",
+            "License :: OSI Approved :: MIT License",
+            "Programming Language :: Python :: 3",
+            "Programming Language :: Python :: 3.4",
```
[the remark this answers is not in the document]
Comment (suzil) [reply setup.py]: Good point, if we use f-strings we must require Python 3.6+. I'll also add a `python_requires` field to enforce this.

File: src/lieu/api.py [hunk 7]
```diff
@@ -29,32 +29,33 @@ def __le__(self, other):
...
-                                                   '' if self.same_phone_number is None else 'same_phone_number={}'.format(self.same_phone_number))
-
-
+        return "Dupe(status={}, sim={}{})".format(
+            self.status.name,
+            self.sim,
+            ""
+            if self.same_phone_number is None
+            else "same_phone_number={}".format(self.same_phone_number),
+        )
```
[the remark this answers is not in the document]
Comment (suzil) [reply src/lieu/api.py]: Changed all .format() to f-string.

File: setup.py [hunk 8]
```diff
@@ -3,43 +3,42 @@
 
 def main():
     setup(
-        name='lieu',
-        version='1.2',
+        name="lieu",
+        version="1.2",
```
Comment (ddelange) [inline setup.py]: bump? semver? move req and version to proper places like rest of our repos?

File: src/lieu/coordinates.py [hunk 9]
```diff
@@ -78,9 +93,9 @@ def is_valid_longitude(longitude):
 
 
 def to_valid_latitude(latitude):
-    '''Convert longitude into the -180 to 180 scale'''
+    """Convert longitude into the -180 to 180 scale"""
```
[the remark this answers is not in the document]
Comment (suzil) [reply src/lieu/coordinates.py]: Checked all docstrings and fixed them to the Google style.

File: setup.py (hunk 8 above)
Comment (suzil) [reply setup.py]: I think keeping version in setup.py is OK? It doesn't seem to make much of a difference here.

File: README.md [hunk 10]
```diff
@@ -1,6 +1,7 @@
 # lieu
 
 [![Build Status](https://travis-ci.com/scoutbee/lieu.svg?branch=master)](https://travis-ci.com/scoutbee/lieu)
+[![black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/python/black)
```
Comment (ddelange) [inline README.md]:
```suggestion
[![black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
```

File: requirements_dev.txt [hunk 11]
```diff
@@ -1,5 +1,6 @@
-autoflake==1.3.1
-boltons==19.1.0
-pre-commit==1.18.3
-pytest==4.6.5
-pytest-sugar==0.9.2
+autoflake
+black
+boltons
```
Comment (ddelange) [inline requirements_dev.txt]: why boltons for dev? used in tests?

Comment (suzil) [reply requirements_dev.txt]: Yup

**Pull request merged**